espnet2/asr/espnet_model.py

```diff
@@ -366,11 +383,13 @@ def _extract_feats(
         # for data-parallel
         speech = speech[:, : speech_lengths.max()]
 
-        if self.frontend is not None:
+        if self.frontend is not None and not self.is_encoder_whisper:
```

Comment:
I think it is enough to just check whether the frontend is None. But you may add an assertion to check if whisper encoder and frontend exist at the same time in the model definition.

Reply:
Ok! this makes sense